### ISSUE TITLE
Missing Java imports for several common classes

### ISSUE THREAD

**Issue by cyrus-and** [ShuhaoZhangTony/briskstream#7]
The commands issued are:

```
mvn validate
mvn install -DskipTests -Dmaven.javadoc.skip=true # <-- errors here
```

Relevant/modified part of `pom.xml`:

```xml
<modules>
    <module>common</module>
    <module>affinity</module>
    <module>BriskBenchmarks</module>
    <module>StormBenchmarks</module>
    <module>FlinkBenchmarks</module>
    <!--<module>HeronBenchmarks</module>-->
</modules>
```

Find the full list below (AFAIK there also could be in the Flink part but maven stopped here):

<details>
<pre>
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/spout/MemFileSpout_Txn.java:[24,15] cannot find symbol
[ERROR]   symbol:   class File
[ERROR]   location: class applications.spout.MemFileSpout_Txn
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/spout/MemFileSpout_Txn.java:[31,25] cannot find symbol
[ERROR]   symbol:   class BufferedWriter
[ERROR]   location: class applications.spout.MemFileSpout_Txn
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/spout/MemFileSpout_Txn.java:[145,88] cannot find symbol
[ERROR]   symbol:   class FileNotFoundException
[ERROR]   location: class applications.spout.MemFileSpout_Txn
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/spout/MemFileSpout_Txn.java:[150,72] cannot find symbol
[ERROR]   symbol:   class FileNotFoundException
[ERROR]   location: class applications.spout.MemFileSpout_Txn
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/spout/MemFileSpout_Txn.java:[166,71] cannot find symbol
[ERROR]   symbol:   class FileNotFoundException
[ERROR]   location: class applications.spout.MemFileSpout_Txn
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/spout/MemFileSpout_latency.java:[24,15] cannot find symbol
[ERROR]   symbol:   class File
[ERROR]   location: class applications.spout.MemFileSpout_latency
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/spout/MemFileSpout_latency.java:[31,25] cannot find symbol
[ERROR]   symbol:   class BufferedWriter
[ERROR]   location: class applications.spout.MemFileSpout_latency
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/spout/MemFileSpout_latency.java:[148,88] cannot find symbol
[ERROR]   symbol:   class FileNotFoundException
[ERROR]   location: class applications.spout.MemFileSpout_latency
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/spout/MemFileSpout_latency.java:[153,72] cannot find symbol
[ERROR]   symbol:   class FileNotFoundException
[ERROR]   location: class applications.spout.MemFileSpout_latency
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/spout/MemFileSpout_latency.java:[169,71] cannot find symbol
[ERROR]   symbol:   class FileNotFoundException
[ERROR]   location: class applications.spout.MemFileSpout_latency
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/bolts/lr/AccidentDetectionBolt.java:[68,19] cannot find symbol
[ERROR]   symbol:   class Map
[ERROR]   location: class applications.bolts.lr.AccidentDetectionBolt
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/bolts/lr/AccidentDetectionBolt.java:[68,32] cannot find symbol
[ERROR]   symbol:   class List
[ERROR]   location: class applications.bolts.lr.AccidentDetectionBolt
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/bolts/lr/AccidentDetectionBolt.java:[72,19] cannot find symbol
[ERROR]   symbol:   class Map
[ERROR]   location: class applications.bolts.lr.AccidentDetectionBolt
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/bolts/lr/AccidentDetectionBolt.java:[72,43] cannot find symbol
[ERROR]   symbol:   class Set
[ERROR]   location: class applications.bolts.lr.AccidentDetectionBolt
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/bolts/lr/AccidentDetection_latencyBolt.java:[70,19] cannot find symbol
[ERROR]   symbol:   class Map
[ERROR]   location: class applications.bolts.lr.AccidentDetection_latencyBolt
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/bolts/lr/AccidentDetection_latencyBolt.java:[70,32] cannot find symbol
[ERROR]   symbol:   class List
[ERROR]   location: class applications.bolts.lr.AccidentDetection_latencyBolt
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/bolts/lr/AccidentDetection_latencyBolt.java:[74,19] cannot find symbol                                                                   
[ERROR]   symbol:   class Map
[ERROR]   location: class applications.bolts.lr.AccidentDetection_latencyBolt
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/bolts/lr/AccidentDetection_latencyBolt.java:[74,43] cannot find symbol                                                                   
[ERROR]   symbol:   class Set
[ERROR]   location: class applications.bolts.lr.AccidentDetection_latencyBolt
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/spout/MemFileSpout.java:[24,15] cannot find symbol                                                                                       
[ERROR]   symbol:   class File
[ERROR]   location: class applications.spout.MemFileSpout
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/spout/MemFileSpout.java:[31,25] cannot find symbol                                                                                       
[ERROR]   symbol:   class BufferedWriter
[ERROR]   location: class applications.spout.MemFileSpout
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/spout/MemFileSpout.java:[145,88] cannot find symbol                                                                                      
[ERROR]   symbol:   class FileNotFoundException
[ERROR]   location: class applications.spout.MemFileSpout
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/spout/MemFileSpout.java:[150,72] cannot find symbol                                                                                      
[ERROR]   symbol:   class FileNotFoundException
[ERROR]   location: class applications.spout.MemFileSpout
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/spout/MemFileSpout.java:[166,71] cannot find symbol                                                                                      
[ERROR]   symbol:   class FileNotFoundException
[ERROR]   location: class applications.spout.MemFileSpout
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/datatypes/toll/FileTollDataStore.java:[39,19] cannot find symbol                                                                         
[ERROR]   symbol:   class File
[ERROR]   location: class applications.datatypes.toll.FileTollDataStore
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/datatypes/toll/FileTollDataStore.java:[48,39] cannot find symbol                                                                         
[ERROR]   symbol:   class IOException
[ERROR]   location: class applications.datatypes.toll.FileTollDataStore
[ERROR] /home/user/briskstream/StormBenchmarks/src/main/java/applications/datatypes/toll/FileTollDataStore.java:[57,30] cannot find symbol                                                                         
[ERROR]   symbol:   class File
[ERROR]   location: class applications.datatypes.toll.FileTollDataStore
</pre>
</details>

Please let me know if you need additional information.

**Comment by cyrus-and** [comment]
Apparently the culprit is 7a4d2070b434337478eb501d4eadff3bb549bed4.

**Comment by ShuhaoZhangTony** [comment]
Seems you have pulled a wrong branch. Try  “stable” branch. Thanks

**Comment by cyrus-and** [comment]
Thanks for the reply. The branch is correct though:

https://github.com/Xtra-Computing/briskstream/blob/d3489a5c4388d5d137aafa8a49d26625cd174402/StormBenchmarks/src/main/java/applications/spout/MemFileSpout_Txn.java#L1-L19

As you can see the import is missing for `File`:

https://github.com/Xtra-Computing/briskstream/blob/d3489a5c4388d5d137aafa8a49d26625cd174402/StormBenchmarks/src/main/java/applications/spout/MemFileSpout_Txn.java#L24

**Comment by cyrus-and** [comment]
I'd be happy to fix that issues myself and then submit a PR, but after fixing them, some more others (this time regarding project classes and not Java common classes) appear and Flink too has similar issues.

I had to do it all by hand because importing the project in IDEA barely works and I thus obtain no support for fixing that errors. But the latter could be a problem on my side as I'm not too familiar with that IDE.

Moreover there are some more road blockers if one decides to build this project on a *modern* distro (say, Debian Buster):

1. the already reported `Makefile` permission (see #8);
2. the fact that this project uses `javah`, a tool removed since JDK 10 (the modern solution is to use `javac -h` and pass source files instead of classes);
3. there are invalid or no more valid SO library search path, for example `$JAVA_HOME/jre/lib/amd64` instead of `$JAVA_HOME/lib/server/`.

More detailed instructions about how to reproduce these benchmarks would be very useful. And provide a Dockerfile that researchers can use to obtain a ready-to-use setup would also be a nice addition.

**Comment by cyrus-and** [comment]
You can reproduce my issues (and see the workarounds) with this [Dockerfile][1]. Watch the build fail with:

```
docker build . -t briskstream-benchmarks
```

[1]: https://gist.github.com/cyrus-and/9c0c3bee40faa86d1f3384ad0493d9bb#file-dockerfile

**Comment by ShuhaoZhangTony** [comment]
Cool, you are very welcomed to PR!

Regarding the JDK. I believe the project only supports JDK8 because the  **_Unsafe_**  API is removed from JDK9 onward. That's very much unfortunate through... :(
Not sure if there is some workaround to resolve it.

The tested environment is **Ubuntu** and **Mac OS**, I haven't tested it on other platforms.

**Comment by cyrus-and** [comment]
> Regarding the JDK. I believe the project only supports JDK8 because the **_Unsafe_** API is removed from JDK9 onward. That's very much unfortunate through... :(

AFAIK it's only a warning so far. The only (solvable) problems are: `javah` and the missing imports.

> Cool, you are very welcomed to PR!

Even if it sound obvious could you please tell me how to import and build the project in IntelliJ IDEA? Simply importing the main directory and letting it pick the `pom.xml` is not enough as errors in sub projects are not considered part of the workspace.

**Comment by ShuhaoZhangTony** [comment]
Sorry I didn't get it, what was the error? 

If you use a clean machine with Ubuntu installed, clone the project and simply let IDEA to "open" (via a GUI) the project (select the folder you just cloned to) should be enough. Besides, you need to make sure the JDK (recommend JDK8) is correctly configured. 

I never use docker/other Linux distributes before, but I guess it should be similar.

**Comment by cyrus-and** [comment]
> Sorry I didn't get it, what was the error?

**Comment by ShuhaoZhangTony** [comment]
BTW: Make sure you use a clean clone of "stable" branch (delete whatever you have). The last error I see from your post still contains obsolete code that causing missing import error.

**Comment by ShuhaoZhangTony** [comment]
That's my bad that I need to apologize: there are multiple actively developing versions of this project that mess things up. The "stable" branch itself should be cleaned enough. But if you have already cloned other versions before (from your previous post, I guess you have), you need to clean them first (some files are not tractable and will not be automatically cleaned when you pull "stable" branch).
If you have already made some modifications, I suggest you back them up, and do a clean clone and migrate your changes to the clean code.

**Comment by cyrus-and** [comment]
I think you should double check what's in your stable (default) branch:

```console
$ git clone https://github.com/Xtra-Computing/briskstream
Cloning into 'briskstream'...
remote: Enumerating objects: 1137, done.
remote: Counting objects: 100% (1137/1137), done.
remote: Compressing objects: 100% (810/810), done.
remote: Total 11095 (delta 420), reused 778 (delta 277), pack-reused 9958
Receiving objects: 100% (11095/11095), 7.90 MiB | 4.25 MiB/s, done.
Resolving deltas: 100% (6550/6550), done.
$ cd briskstream/
$ git checkout stable 
Already on 'stable'
Your branch is up to date with 'origin/stable'.
$ head -24 StormBenchmarks/src/main/java/applications/spout/MemFileSpout_Txn.java
package applications.spout;


import constants.BaseConstants;
import helper.wrapper.StringStatesWrapper;
import util.OsUtils;
import org.apache.storm.tuple.Values;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.lang.management.ManagementFactory;
import java.lang.management.RuntimeMXBean;
import java.util.LinkedList;
import java.util.List;
import java.util.Scanner;

/**
 * @author Maycon Viana Bordin <mayconbordin@gmail.com>
 */
public class MemFileSpout_Txn extends AbstractSpout {
    private static final Logger LOG = LoggerFactory.getLogger(MemFileSpout_Txn.class);
    private static LinkedList<String> logger = new LinkedList<>();
    protected final String splitregex = ",";
    protected File files;
```

As you can see, for example, no `import java.io.File;`.

**Comment by ShuhaoZhangTony** [comment]
ah, I see! 
Storm and Flink submodules are outdated. I didn't find them outdated because I commented them out.
I will update them and fix those error very soon. Thanks for let me know!

**Comment by cyrus-and** [comment]
Awesome, thanks! :)

**Comment by ShuhaoZhangTony** [comment]
Done! Let me know if it works on your site.

**Comment by cyrus-and** [comment]
Thank you, it now seems to compile just fine, except for the Makefile issue that is fixed in #9.
